core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java

```diff
@@ -193,7 +193,9 @@ private void initMeasureChunkIndexes() {
   public BitSetGroup applyFilter(RawBlockletColumnChunks rawBlockletColumnChunks,
       boolean useBitsetPipeLine) throws FilterUnsupportedException, IOException {
     if (exp instanceof MatchExpression) {
-      return rawBlockletColumnChunks.getBitSetGroup();
+      // if there are datamap created on this table, it should not come to here
+      throw new FilterUnsupportedException(
+          exp.getFilterExpressionType().name() + " is not supported on this table");
```

Comment:
Won't it be better to add tableName information to the error message, such as
```
... is not supported on $dbName.$tableName
```

Reply:
ok. fixed

Reply:
not yet, need to add doc